File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string str;
        for(int i = 0; i < s.size(); i++){
            if(s[i] >= 'a' && s[i] <= 'z' || s[i] >= '0' && s[i] <= '9'){
                str.push_back(s[i]);
            }
            else if(s[i] >= 'A' && s[i] <= 'Z'){
                str.push_back(s[i]+32);
            }
            else{
                continue;
            }
        }
        int start = 0;
        int end = str.size()-1;
        while(start <= end){
            if(str[start] != str[end]){
                return false;
            }
            else{
                start++;
                end--;
            }
        }
        return true;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.py
# class Solution:
#     def isPalindrome(self, s: str) -> bool:
#         clean = ''.join(c.lower() for c in s if c.isalnum())
#         return clean[::] == clean[::-1]


class Solution:
    def checkPalindrome(self, arr, i, j):
        if i >= j:
            return True

        if arr[i] != arr[j]:
            return False
        
        return self.checkPalindrome(arr, i+1, j-1)


    def isPalindrome(self, s: str) -> bool:
        clean = ''.join(c.lower() for c in s if c.isalnum())
        i = 0
        j = len(clean)-1
        return self.checkPalindrome(clean, i, j)
        